### ISSUE TITLE
Pressing Enter in MultiAdmin 3.0.0 occasionally causes server crashes

### ISSUE THREAD

**Issue by PatPeter** [Grover-c13/MultiAdmin#207]
A continuation of the MA 2.0 bug that crashed servers, or maybe it was never fixed from one version to the next:

```c#
Unhandled Exception:
System.ArgumentOutOfRangeException: Value must be positive and below the buffer height.
Parameter name: top
  at System.TermInfoDriver.SetCursorPosition (System.Int32 left, System.Int32 top) <0x7fe05a94e080 + 0x0018f> in <61d5ec988156494684532710ebf
1643d>:0
  at System.ConsoleDriver.SetCursorPosition (System.Int32 left, System.Int32 top) <0x7fe05a940120 + 0x00036> in <61d5ec988156494684532710ebf1
643d>:0
  at System.Console.SetCursorPosition (System.Int32 left, System.Int32 top) <0x7fe05a93f0f0 + 0x00005> in <61d5ec988156494684532710ebf1643d>:0
  at MultiAdmin.MultiAdmin.InputThread.Write (MultiAdmin.MultiAdmin.Server server) [0x00040] in <bbb11ca17cb9463383a3c777d3dd6ef4>:0
  at MultiAdmin.MultiAdmin.Server.<.ctor>b__60_0 () [0x00000] in <bbb11ca17cb9463383a3c777d3dd6ef4>:0
  at System.Threading.ThreadHelper.ThreadStart_Context (System.Object state) <0x7fe05aa14eb0 + 0x00084> in <61d5ec988156494684532710ebf1643d>:0
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) <0x7fe05aa12e80 + 0x00171> in <61d5ec988156494684532710ebf1643d>:0
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state) <0x7fe05aa12e30 + 0x00016> in <61d5ec988156494684532710ebf1643d>:0
  at System.Threading.ThreadHelper.ThreadStart () <0x7fe05aa14fc0 + 0x00020> in <61d5ec988156494684532710ebf1643d>:0
[ERROR] FATAL UNHANDLED EXCEPTION: System.ArgumentOutOfRangeException: Value must be positive and below the buffer height.
Parameter name: top
  at System.TermInfoDriver.SetCursorPosition (System.Int32 left, System.Int32 top) <0x7fe05a94e080 + 0x0018f> in <61d5ec988156494684532710ebf1643d>:0
  at System.ConsoleDriver.SetCursorPosition (System.Int32 left, System.Int32 top) <0x7fe05a940120 + 0x00036> in <61d5ec988156494684532710ebf1643d>:0
  at System.Console.SetCursorPosition (System.Int32 left, System.Int32 top) <0x7fe05a93f0f0 + 0x00005> in <61d5ec988156494684532710ebf1643d>:0
  at MultiAdmin.MultiAdmin.InputThread.Write (MultiAdmin.MultiAdmin.Server server) [0x00040] in <bbb11ca17cb9463383a3c777d3dd6ef4>:0
  at MultiAdmin.MultiAdmin.Server.<.ctor>b__60_0 () [0x00000] in <bbb11ca17cb9463383a3c777d3dd6ef4>:0
  at System.Threading.ThreadHelper.ThreadStart_Context (System.Object state) <0x7fe05aa14eb0 + 0x00084> in <61d5ec988156494684532710ebf1643d>:0
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) <0x7fe05aa12e80 + 0x00171> in <61d5ec988156494684532710ebf1643d>:0
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state) <0x7fe05aa12e30 + 0x00016> in <61d5ec988156494684532710ebf1643d>:0
  at System.Threading.ThreadHelper.ThreadStart () <0x7fe05aa14fc0 + 0x00020> in <61d5ec988156494684532710ebf1643d>:0
```

**Comment by Dankrushen** [comment]
You're not running MultiAdmin v3

**Comment by Dankrushen** [comment]
I take that back again, that really really doesn't seem to be MultiAdmin v3; it looks like you still have v2, can you make sure you have v3?

**Comment by Dankrushen** [comment]
Closing due to inactivity, please re-open if you find the issue is still happening.

**Comment by PatPeter** [comment]
It happens rarely and now I just don't type in my server console so that I don't take the risk. If I ever have time I'll look into it.

**Comment by PatPeter** [comment]
@Dankrushen I figured out how to reproduce this!!!

1. Connect to server via PuTTY.
2. Run server in screen:

    screen -R -h ${SCROLLBACK:-10240} -S scp1 mono MultiAdmin.exe

3. Resize the screen.
4. Press Enter.
5. The server will crash.

Please reopen this issue.

**Comment by Dankrushen** [comment]
Can you please post the new exception?

**Comment by PatPeter** [comment]
It's a little corrupted.

```
[14:42:38] >>> 717;1R
[14:42:38] Sending request to SCP: Secret Laboratory...
[14:42:36] >>>
Unhandled Exception:equest to SCP: Secret Laboratory...SCP Secret Laboratory/config_gameplay.txt
System.ArgumentOutOfRangeException: Value must be positive and below the buffer height.
Parameter name: topeam/scp/servers/scp7/config.txt
  at System.TermInfoDriver.SetCursorPosition (System.Int32 left, System.Int32 top) <0x7f456414e080 + 0x0018f> in <61d5ec988156494684532710ebf1643d>:0 ession.ers/scp8/config.txt
  at System.ConsoleDriver.SetCursorPosition (System.Int32 left, System.Int32 top) <0x7f4564140120 + 0x00036> in <61d5ec988156494684532710ebf1643d>:0 swapped: config.txt, config_custom.txt, config_gameplay.txt, config_remoteadmin.txt, sm_config_gamemode  at System.Console.SetCursorPosition (System.Int32 left, System.Int32 top) <0x7f456413f0f0 + 0x00005> in <61d5ec988156494684532710ebf1643d>:0 ng server with the following parameters
  at MultiAdmin.MultiAdmin.InputThread.Write (MultiAdmin.MultiAdmin.Server server) [0x00040] in <bbb11ca17cb9463383a3c777d3dd6ef4>:0 19-04-20_14_42_SCP_output_log.txt"
  at MultiAdmin.MultiAdmin.Server.<.ctor>b__60_0 () [0x00000] in <bbb11ca17cb9463383a3c777d3dd6ef4>:0 and SlenderPlays.
  at System.Threading.ThreadHelper.ThreadStart_Context (System.Object state) <0x7f4564214eb0 + 0x00084> in <61d5ec98815649468453
2710ebf1643d>:0  request LocalAdmin source code at moszka.hubert@gmail.com.
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) <0x7f4564212e80 + 0x00171> in <61d5ec988156494684532710ebf1643d>:0
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state) <0x7f4564212e30 + 0x00016> in <61d5ec988156494684532710ebf1643d>:0
  at System.Threading.ThreadHelper.ThreadStart () <0x7f4564214fc0 + 0x00020> in <61d5ec988156494684532710ebf1643d>:0
[ERROR] FATAL UNHANDLED EXCEPTION: System.ArgumentOutOfRangeException: Value must be positive and below the buffer height.
Parameter name: top
  at System.TermInfoDriver.SetCursorPosition (System.Int32 left, System.Int32 top) <0x7f456414e080 + 0x0018f> in <61d5ec988156494684532710ebf1643d>:0
  at System.ConsoleDriver.SetCursorPosition (System.Int32 left, System.Int32 top) <0x7f4564140120 + 0x00036> in <61d5ec988156494684532710ebf1643d>:0
  at System.Console.SetCursorPosition (System.Int32 left, System.Int32 top) <0x7f456413f0f0 + 0x00005> in <61d5ec988156494684532710ebf1643d>:0
  at MultiAdmin.MultiAdmin.InputThread.Write (MultiAdmin.MultiAdmin.Server server) [0x00040] in <bbb11ca17cb9463383a3c777d3dd6ef4>:0
  at MultiAdmin.MultiAdmin.Server.<.ctor>b__60_0 () [0x00000] in <bbb11ca17cb9463383a3c777d3dd6ef4>:0
  at System.Threading.ThreadHelper.ThreadStart_Context (System.Object state) <0x7f4564214eb0 + 0x00084> in <61d5ec988156494684532710ebf1643d>:0
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) <0x7f4564212e80 + 0x00171> in <61d5ec988156494684532710ebf1643d>:0
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state) <0x7f4564212e30 + 0x00016> in <61d5ec988156494684532710ebf1643d>:0
  at System.Threading.ThreadHelper.ThreadStart () <0x7f4564214fc0 + 0x00020> in <61d5ec988156494684532710ebf1643d>:0
Press Enter to stop the server...
```

**Comment by Dankrushen** [comment]
That doesn't look right, are you sure you're on the latest MultiAdmin?

**Comment by PatPeter** [comment]
So I tried upgrading to MultiAdmin 3.1.2 and now I'm having trouble with every single version of MultiAdmin that I try. Two of my servers are hosed. I'm going to downgrade to 2.1.2 until I can find out where these issues are coming from.

**Comment by PatPeter** [comment]
It turns out my Continuous Integration script was pushing production settings to my test environment and constantly overriding MultiAdmin 3.x with MultiAdmin 2.x. Sorry but when I saw this error again the last thing I wanted was for this to have made it into MultiAdmin 3.x.

Thank you so much for working so hard on MultiAdmin 3.x! Now that I've figured out how to get around the ways I had my repositories set up in conflict with MultiAdmin 3.x, it's been a lot easier to work with. (Also the version I had issues on was 3.0.0) I LOVE the new input system! Being able to press Enter without an annoying response is the best 💯
